testing_and_setup/compass/ocean/global_ocean/CUSP12/init/define_base_mesh.py

```diff
@@ -1,187 +1,375 @@
-#!/usr/bin/env python
-from __future__ import absolute_import, division, print_function, \
-    unicode_literals
-
+import cartopy.crs as ccrs
+from cartopy import config
+import matplotlib.pyplot as plt
 import numpy as np
-import jigsaw_to_MPAS.coastal_tools as ct
+import jigsaw_to_MPAS.mesh_definition_tools as mdt
+from jigsaw_to_MPAS.coastal_tools import signed_distance_from_geojson, \
+    mask_from_geojson
+from geometric_features import read_feature_collection
+import xarray
+import matplotlib
+import matplotlib.colors as colors
+matplotlib.use('Agg')
+
+def truncate_colormap(cmapIn='jet', minval=0.0, maxval=1.0, n=100):
+    '''truncate_colormap(cmapIn='jet', minval=0.0, maxval=1.0, n=100)'''    
+    cmapIn = plt.get_cmap(cmapIn)
+
+    new_cmap = colors.LinearSegmentedColormap.from_list(
+        'trunc({n},{a:.2f},{b:.2f})'.format(n=cmapIn.name, a=minval, b=maxval),
+        cmapIn(np.linspace(minval, maxval, n)))
+
+    return new_cmap
+
+def plot_cartopy(nPlot, varName, var, cmapIn):
+    ax = plt.subplot(6, 2, nPlot, projection=ccrs.PlateCarree())
+    ax.set_global()
+    # plot in units of km
+    if np.amax(var)<2:
+        pltvar = var
+    else:
+        pltvar = var/1000.
+    im = ax.imshow(pltvar,
+                   origin='lower',
+                   transform=ccrs.PlateCarree(),
+                   extent=[-180, 180, -90, 90], 
+                   cmap=cmapIn)
+    ax.coastlines()
+    gl = ax.gridlines(
+        crs=ccrs.PlateCarree(),
+        draw_labels=True,
+        linewidth=1,
+        color='gray',
+        alpha=0.5,
+        linestyle='-')
+    gl.xlabels_top = False
+    gl.xlabels_bottom = False
+    gl.ylabels_right = False
+    gl.ylabels_left = False
+    plt.colorbar(im, shrink=.9)
+    plt.title(varName)
 
 
 def cellWidthVsLatLon():
+    """
+    Create cell width array for this mesh on a regular latitude-longitude grid.
+    Returns
+    -------
+       cellWidth : numpy.ndarray
+            m x n array, entries are desired cell width in km
+       lat : numpy.ndarray
+            latitude, vector of length m, with entries between -90 and 90,
+            degrees
+       lon : numpy.ndarray
+            longitude, vector of length n, with entries between -180 and 180,
+            degrees
+    """
+    # To speed up for testing, set following line to 1.0 degrees
+    dlon = 1.0
+    dlat = dlon
+    nlon = int(360. / dlon) + 1
+    nlat = int(180. / dlat) + 1
+    lon = np.linspace(-180., 180., nlon)
+    lat = np.linspace(-90., 90., nlat)
+    km = 1.0e3
+
+    # prepare plots
+    fig = plt.figure()
+    plt.clf()
+    fig.set_size_inches(10.0, 14.0)
+    cmapBluesHalf = truncate_colormap(cmapIn='Blues', minval=0.0, maxval=0.7)
+
+    # Create cell width vs latitude for Atlantic and Pacific basins
+    QU1 = np.ones(lat.size)
+    EC60to30 = mdt.EC_CellWidthVsLat(lat)*km
+    EC60to30Narrow = mdt.EC_CellWidthVsLat(lat, latPosEq = 8.0, latWidthEq = 3.0)*km
```

Comment:
@vanroekel seems to think in #522 that we want the equator region to be at least 10 degrees wide (probably with a 4.0 degree transition for consistency).  I think we want to be consistent between all the meshes, so this should probably do the same, as should #518